### PR TITLE
Allow to override Integration operator references

### PR DESCRIPTION
**Pull request by davidmogar**
This commits allows to override the Integration operator image by setting the new image and tag in the INTEGRATION_IMAGE_REPO and INTEGRATION_IMAGE_TAG variables. It also allows to override the path to the resources through the INTEGRATION_RESOURCES variable.

Signed-off-by: David Moreno García <damoreno@redhat.com>

### CODE REVIEW

File: hack/preview.sh [hunk 1]
```diff
@@ -89,6 +89,9 @@ fi
...
 [ -n "${HAS_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/application-service\")) |=.newTag=\"${HAS_IMAGE_TAG}\"" $ROOT/components/has/kustomization.yaml
 [[ -n "${HAS_PR_OWNER}" && "${HAS_PR_SHA}" ]] && yq -i e "(.resources[] | select(. ==\"*github.com/redhat-appstudio/application-service*\")) |= \"https://github.com/${HAS_PR_OWNER}/application-service/config/default?ref=${HAS_PR_SHA}\"" $ROOT/components/has/kustomization.yaml
+[ -n "${INTEGRATION_IMAGE_REPO}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/integration-service\")) |=.newName=\"${INTEGRATION_IMAGE_REPO}\"" $ROOT/components/integration/kustomization.yaml
+[ -n "${INTEGRATION_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/integration-service\")) |=.newTag=\"${INTEGRATION_IMAGE_TAG}\"" $ROOT/components/integration/kustomization.yaml
+[ -n "${INTEGRATION_RESOURCES}" ] && yq -i e "(.resources[] | select(.==\"https://github.com/redhat-appstudio/integration-service/config/default?ref=*\")) |=.=\"${INTEGRATION_RESOURCES}\"" components/integration/kustomization.yaml
```
Comment (Michkov) [inline hack/preview.sh:94]: Missing $ROOT

Comment (davidmogar) [reply hack/preview.sh]: It seems this is coming from an early mistake I made when adding the RELEASE_* variables. I fixed both here.

File: hack/preview.sh [hunk 2]
```diff
@@ -89,9 +89,12 @@ fi
...
 [ -n "${HAS_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/application-service\")) |=.newTag=\"${HAS_IMAGE_TAG}\"" $ROOT/components/has/kustomization.yaml
 [[ -n "${HAS_PR_OWNER}" && "${HAS_PR_SHA}" ]] && yq -i e "(.resources[] | select(. ==\"*github.com/redhat-appstudio/application-service*\")) |= \"https://github.com/${HAS_PR_OWNER}/application-service/config/default?ref=${HAS_PR_SHA}\"" $ROOT/components/has/kustomization.yaml
+[ -n "${INTEGRATION_IMAGE_REPO}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/integration-service\")) |=.newName=\"${INTEGRATION_IMAGE_REPO}\"" $ROOT/components/integration/kustomization.yaml
+[ -n "${INTEGRATION_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/integration-service\")) |=.newTag=\"${INTEGRATION_IMAGE_TAG}\"" $ROOT/components/integration/kustomization.yaml
+[ -n "${INTEGRATION_RESOURCES}" ] && yq -i e "(.resources[] | select(.==\"https://github.com/redhat-appstudio/integration-service/config/default?ref=*\")) |=.=\"${INTEGRATION_RESOURCES}\"" $ROOT/components/integration/kustomization.yaml
```
Comment (Michkov) [inline hack/preview.sh:94]: integration service is not defined in components yet. This does not provide any value and cannot be tested.

Comment (davidmogar) [reply hack/preview.sh:94]: Yes, that is going to come soon and will be added by @dirgim. You can keep this open (or on draft) till it's done.

**Review by dirgim — APPROVED** [review]
lgtm

File: hack/preview.sh (hunk 2 above)
Comment (dirgim) [reply hack/preview.sh:94]: The integration component has been added, so this PR is now relevant for testing integration service changes.

**Pull request merged**